changing config

nm3224 opened the pull request:
## changes & context

### Config Changes: 
- adding default filtersfor retention config
      - core terms only (can be adjusted) 
      - associate's students only
      - first time admits only 
- changing incomplete threshold from 0.5 to 0.2
- added extra parameters with a TODO to edit based on school's preferences for checkpoint; default is always core terms only and excluding any pre-cohort terms _unless specified otherwise_
     - exclude_non_core_terms = true
     - exclude_pre_cohort_terms = true

### Template Changes:
- adding gateway course cip codes as a default too in the template
- adding prints for top 10 and bottom 10 support scores for slide deck visibility
- adding catalog variable so only have to change in one place 

### Checkpointing Logic Changes:
- changing checkpointing code for retention to default as excluding non-core terms and pre-cohort terms alike
- adding descriptive sentences to args missing them

### Feature Table Changes:
- adding "number of" at the start and "taken this term" at the end

Commit:
changing config

Commit:
updating config for math/english cips

Commit:
adjusting for linter

Commit:
template updates

- adding prints for top 10 and bottom 10 support scores for slide deck visibility
- changing incomplete threshold from 0.5 to 0.8
- changing catalog from staging_sst_01 to sst_dev

Commit:
adding note to use core terms too

Commit:
fixing for linter

nm3224 committed:
Merge branch 'develop' into retention_config_update

Commit:
updating feature table

kaylawilding commented on pipelines/pdp/institution_id/config-CREDITS_EARNED_TEMPLATE.toml:
```diff
@@ -65,7 +65,7 @@ num_terms_in_year = 4
 
 [modeling.feature_selection]
-incomplete_threshold = 0.5
+incomplete_threshold = 0.8
```
We want to drop features with 20% or more missing correct? The function that uses this drops features with greater than or equal to the threshold missing values. I think this should be 0.2. Double check the fn defn for drop_incomplete_features in feature selection though please. Also, we should probably change the default in that function as well from 0.5. If Im understanding this correctly, please make the same edit to the other config templates

nm3224 replied:
oh shoot, i thought it was we want to drop features with 80% or more missing... @vishpillai123 am i understanding this wrong

nm3224 replied:
@kaylawilding i'll check on this and change as needed!

nm3224 replied:
changed! I think you're right @kaylawilding

kaylawilding commented on pipelines/pdp/institution_id/config-RETENTION_TEMPLATE.toml:
```diff
@@ -42,7 +42,8 @@ key_course_subject_areas = []
 [preprocessing.selection]
-student_criteria = { enrollment_type = "", credential_type_sought_year_1 = "" }
+# default is usually fall-to-fall retention, if otherwise then change to core_terms
+student_criteria = { enrollment_type = "", credential_type_sought_year_1 = "", cohort_term = "FALL" }
```
Lets set this up for the default models? We can chat about what should the default model params be in our 1-1 if that helps

nm3224 replied:
sounds good!

kaylawilding commented on pipelines/pdp/institution_id/03-make-predictions-TEMPLATE.py:
```diff
@@ -273,12 +279,14 @@
 
 with mlflow.start_run() as run:
     confusion_matrix = evaluation.log_confusion_matrix(
+        catalog="sst_dev",
```
actually can you add a variable for catalog and then make these four instance call that var - just so we only have to edit one place instead of 4 if it needs changed?

nm3224 replied:
yup!

kaylawilding commented on pipelines/pdp/institution_id/01-preprocess-data-TEMPLATE.py:
```diff
@@ -135,6 +136,11 @@
 # COMMAND ----------
 
+# confirm cip code list, then add to config
+cip_code_list = ["23", "27"]
```
sorry whats happening here? these are the cip codes for the math and english gateways? the user needs to confirm these are correct for the school and then add to the config?

nm3224 replied:
yes correct!

nm3224 replied:
it typically is 23 and 27 for like all the schools i've worked on, but user shud always confirm

Commit:
making requested changed

- adding catalog variable so only have to change in one place
- adjusting incomplete threshold from 0.8 to 0.2
- changing checkpointing code for retention to default exclude non-core terms and pre-cohort terms alike

Commit:
fixed duplicate

Commit:
fixing linting

Commit:
fixing unit tests

Commit:
fixed index

Commit:
changed unit test

Commit:
fixed unit test

Commit:
fixing terms

Commit:
changing include cols param

Commit:
changing exp

Commit:
added "term_is_core"

Commit:
added into unit test

Commit:
fixed length

Commit:
changing to false

Commit:
put true back

Commit:
changing tests

Commit:
added default filters

Commit:
exclude non core terms from default

Commit:
changing order

nm3224 commented on src/student_success_tool/preprocessing/checkpoints/pdp.py:
```diff
@@ -190,15 +190,18 @@ def first_student_terms_at_num_credits_earned(
+    term_is_pre_cohort_col: str = "term_is_pre_cohort",
+    exclude_pre_cohort_terms: bool = True,
+    term_is_core_col: str = "term_is_core",
+    exclude_non_core_terms: bool = True,
```
add to config

nm3224 commented on src/student_success_tool/preprocessing/checkpoints/pdp.py:
```diff
@@ -190,15 +190,18 @@ def first_student_terms_at_num_credits_earned(
     student_id_cols: str | list[str] = "student_id",
     sort_cols: str | list[str] = "term_rank",
     include_cols: t.Optional[list[str]] = None,
+    term_is_pre_cohort_col: str = "term_is_pre_cohort",
```
check if this is redundant

Commit:
adding default params to the config which should be adjusted based on school's preference

Commit:
added a note in pre processing

vishpillai123 commented on pipelines/pdp/institution_id/03-make-predictions-TEMPLATE.py:
```diff
@@ -308,7 +319,7 @@
 dataio.write.to_delta_table(
     shap_feature_importance,
-    f"staging_sst_01.{cfg.institution_id}_silver.training_{cfg.model.run_id}_shap_feature_importance",
+    f"sst_dev.{cfg.institution_id}_silver.training_{cfg.model.run_id}_shap_feature_importance",
```
can you add catalog here as well?

nm3224 replied:
done!

vishpillai123 commented on pipelines/pdp/institution_id/03-make-predictions-TEMPLATE.py:
```diff
@@ -328,6 +339,6 @@
 dataio.write.to_delta_table(
     support_score_distribution,
-    f"staging_sst_01.{cfg.institution_id}_silver.training_{cfg.model.run_id}_support_overview",
+    f"sst_dev.{cfg.institution_id}_silver.training_{cfg.model.run_id}_support_overview",
```
and catalog here as well

nm3224 committed:
Merge branch 'develop' into retention_config_update

Commit:
changed to catalog

nm3224 committed:
Merge branch 'develop' into retention_config_update

kaylawilding commented on pipelines/pdp/institution_id/01-preprocess-data-TEMPLATE.py:
```diff
@@ -239,6 +245,7 @@
 
 # TODO: choose checkpoint function suitable for school's use case
 # parameters should be specified in the config
+# MAKE SURE TO SPECIFY exclude_non_core_terms and exclude_pre_cohort_terms per school's needs!
```
👍

kaylawilding commented on pipelines/pdp/institution_id/config-RETENTION_TEMPLATE.toml:
```diff
@@ -42,19 +42,23 @@ key_course_subject_areas = []
 [preprocessing.selection]
-student_criteria = { enrollment_type = "", credential_type_sought_year_1 = "" }
+# if fall-to-fall retention, change cohort_term = "FALL"
+student_criteria = { enrollment_type = "FIRST-TIME", credential_type_sought_year_1 = "ASSOCIATE'S DEGREE", cohort_term = ["FALL", "SPRING"] }
```
👍

kaylawilding commented on src/student_success_tool/preprocessing/checkpoints/pdp.py:
```diff
@@ -190,15 +190,18 @@ def first_student_terms_at_num_credits_earned(
 def first_student_terms_within_cohort(
```
This changes the default from what older schools would have used, correct? Can you just put this on your list to document in confluence this week?

nm3224 replied:
yes- they would just have to set `exclude_non_core_terms` to false for it to be the same as before (we were alr removing pre-cohort anyway)

nm3224 replied:
@kaylawilding where in the confluence should the change go / shud i slack the team ab this in our channel

kaylawilding replied:
ya just slack me or I think I made you an asana task - just tag me there please :)

kaylawilding commented on src/student_success_tool/preprocessing/targets/pdp/graduation.py:
```diff
@@ -117,10 +117,11 @@ def compute_target(
             sort_cols=term_rank_col,
             include_cols=(student_id_cols + [term_rank_col, enrollment_intensity_col]),
+            term_is_pre_cohort_col=term_is_pre_cohort_col,
+            term_is_core_col="term_is_core",
```
I think ideally in the long-run lets not hardcode this feature but we can put that on the backlog

Commit:
excluding non-core terms to remain consistent with what we've been doing in this function

Commit:
remove space for linting

vishpillai123 commented on pipelines/pdp/institution_id/03-make-predictions-TEMPLATE.py:
```diff
@@ -308,7 +319,7 @@
 dataio.write.to_delta_table(
     shap_feature_importance,
-    f"staging_sst_01.{cfg.institution_id}_silver.training_{cfg.model.run_id}_shap_feature_importance",
+    f"catalog.{cfg.institution_id}_silver.training_{cfg.model.run_id}_shap_feature_importance",
```
this needs to be {catalog} since it's an f-string

kaylawilding replied:
oh good catch

Commit:
adding {} for f'string
